packages/paste-core/components/combobox/src/Combobox.tsx

```diff
@@ -29,104 +29,108 @@ const StyledInputAsSelect = styled(InputElement)<ComboboxProps>(props =>
+            highlightedIndex={highlightedIndex}
+            optionTemplate={optionTemplate}
+            ref={ref}
+          />
+        ))}
+      </>
+    );
+  }
+);
 
-const GroupedItems: React.FC<GroupItemsProps> = ({
-  items,
-  getItemProps,
-  highlightedIndex,
-  optionTemplate,
-  groupLabelTemplate,
-  groupItemsBy,
-}) => {
-  const UIDSeed = useUIDSeed();
+const GroupedItems = React.forwardRef<HTMLDivElement, GroupItemsProps>(
+  ({items, getItemProps, highlightedIndex, optionTemplate, groupLabelTemplate, groupItemsBy}, ref) => {
+    const UIDSeed = useUIDSeed();
 
-  // Should never happen
-  if (groupItemsBy == null) {
-    return null;
-  }
+    // Should never happen
+    if (groupItemsBy == null) {
+      return null;
+    }
 
-  const groupedItems = groupBy(items, (item: ItemType) => item[groupItemsBy]);
-  const groupedItemKeys = Object.keys(groupedItems);
+    const groupedItems = groupBy(items, (item: ItemType) => item[groupItemsBy]);
+    const groupedItemKeys = Object.keys(groupedItems);
 
-  return (
-    <>
-      {groupedItemKeys.map(groupedItemKey => {
-        // These items are categorized as ungrouped
-        if (groupedItemKey === 'undefined') {
-          return groupedItems[groupedItemKey].map((item: ItemType, index: number) => (
-            <Item
-              item={item}
-              index={index}
-              key={UIDSeed(`ungrouped-${index}`)}
-              getItemProps={getItemProps}
-              highlightedIndex={highlightedIndex}
-              optionTemplate={optionTemplate}
-            />
-          ));
-        }
-        return (
-          <ComboboxListboxGroup
-            groupName={groupedItemKey}
-            groupLabelTemplate={groupLabelTemplate}
-            key={UIDSeed(groupedItemKey)}
-          >
-            {groupedItems[groupedItemKey].map((item: ItemType, index: number) => {
-              return (
-                <Item
-                  item={item}
-                  index={UIDSeed(`${groupedItemKey}-${index}`)}
-                  key={UIDSeed(`${groupedItemKey}-${index}`)}
-                  getItemProps={getItemProps}
-                  highlightedIndex={highlightedIndex}
-                  optionTemplate={optionTemplate}
-                  inGroup
-                />
-              );
-            })}
-          </ComboboxListboxGroup>
-        );
-      })}
-    </>
-  );
-};
+    return (
+      <>
+        {groupedItemKeys.map(groupedItemKey => {
+          // These items are categorized as ungrouped
+          if (groupedItemKey === 'undefined') {
+            return groupedItems[groupedItemKey].map((item: ItemType, index: number) => (
+              <Item
+                item={item}
+                index={index}
+                key={UIDSeed(`ungrouped-${index}`)}
+                getItemProps={getItemProps}
+                highlightedIndex={highlightedIndex}
+                optionTemplate={optionTemplate}
+                ref={ref}
```

Comment:
We shouldn't pass the same ref in a map. This file may need more consideration.

Reply:
Right. This ref should only be passed to the `ComboboxListboxGroup`.